packages/eth-json-rpc-provider/src/safe-event-emitter-provider.test-d.ts

```diff
@@ -0,0 +1,22 @@
+import { Web3Provider } from '@ethersproject/providers';
+import EthQuery from '@metamask/eth-query';
+import EthJsQuery from '@metamask/ethjs-query';
+import { JsonRpcEngine } from '@metamask/json-rpc-engine';
+import { BrowserProvider } from 'ethers';
+
+import { SafeEventEmitterProvider } from './safe-event-emitter-provider';
+
+const engine = new JsonRpcEngine();
+const provider = new SafeEventEmitterProvider({ engine });
+
+// /* @metamask/eth-query */
+new EthQuery(provider);
+
+// /* @metamask/ethjs-query */
+new EthJsQuery(provider);
```

Comment:
I realized, after reviewing this full PR, that `@metamask/ethjs-query` doesn't actually have any types. That's why we have to add them below, in `types/ethjs-query.d.ts`.

That means that there's not much point in testing `@metamask/ethjs-query` like this; we're the ones providing the types in this PR, and we're essentially giving a type of `any` to the `EthJsQuery` constructor here, so this line is guaranteed to work — it won't fail.

Because of this, I'm now doubtful whether this is the correct test. Instead, I'm curious whether we ought to do more of a smoke test? In other words, we:

- create a JsonRpcEngine that returns a fake response for a particular RPC method in the Ethereum spec (e.g. `eth_blockNumber`)
- create a provider using that engine
- use the library to make a request for that method
- assert that the response that the library returns matches the response that we had the engine return

Doing this means this file would become a regular test file and we wouldn't need to use `tsd` (if there happens to be a type error, then the tests wouldn't run at all, so we don't need to explicitly test for it).

The smoke test for each library would work like this:

- For `@metamask/eth-query`, we can call [`ethQuery.sendAsync`](https://github.com/MetaMask/eth-query/blob/31320fe5fd776564c018019a526a524290406441/index.js#L111) and pass the RPC method we want to request.
- For `@metamask/ethjs-query`, it doesn't have a `sendAsync` method; it automatically creates an object method for each RPC method it knows about (that list is defined [here](https://github.com/MetaMask/ethjs-schema/blob/69a8f6c14e0bb37d3beed64114af28e1237b67dd/src/schema.json)). That's why we have to have the engine handle an RPC method in the Ethereum spec. So for example we'd call `ethJsQuery.blockNumber()`.
- For `Web3Provider`, we can call `web3Provider.send` and pass the RPC method we want to request (note that Web3Provider inherit from JsonRpcProvider; see [docs](https://docs.ethers.org/v5/api/providers/jsonrpc-provider/)).
- For `BrowserProvider`, we can call `browserProvider.send` and pass the RPC method we want to request (see [docs](https://docs.ethers.org/v6/api/providers/#BrowserProvider)).

What are your thoughts?